tests/Unit/Routes/WebRoutesTest.php

```diff
@@ -1,20 +1,153 @@
+            ->assertResponseStatus(200)
+        ;
+    }
+
+    /** @test */
+    public function testDashboardRouteGate()
+    {
+        $route = URL::route('service.dashboard');
+
+        Auth::logout();
+        // You cannot get there logged out.
+        $this->visit($route)
+            ->seePageIs(URL::route('service.login'))
+            ->assertResponseStatus(200)
+        ;
+        // You can get there logged in.
+        $this->actingAs($this->user)
+            ->visit($route)
+            ->seePageIs($route)
+            ->assertResponseStatus(200)
+        ;
+    }
+
+    /** @test */
+    public function testSearchRouteGate()
+    {
+
+        $route = URL::route('service.registration.index');
+
+        Auth::logout();
+        // You cannot get there logged out.
+        $this->visit($route)
+            ->seePageIs(URL::route('service.login'))
+            ->assertResponseStatus(200)
+        ;
+        // You can get there logged in.
+        $this->actingAs($this->user)
+            ->visit($route)
+            ->seePageIs($route)
+            ->assertResponseStatus(200)
+        ;
+    }
+
+    /** @test */
+    public function testViewRouteGate()
     {
-        $this->get(route('service.login'))
-            ->assertResponseStatus(200);
+
+        // Create a random registration with our centre.
+        $registration = factory(App\Registration::class)->create([
+            "centre_id" => $this->centre->id,
+        ]);
+
+        $route = URL::route('service.registration.edit', [ 'registration' => $registration->id ]);
+
+        Auth::logout();
+        // You cannot get there logged out.
+        $this->visit($route)
+            ->seePageIs(URL::route('service.login'))
+            ->assertResponseStatus(200)
+        ;
+        // You can get there logged in.
+        $this->actingAs($this->user)
+            ->visit($route)
+            ->seePageIs($route)
+            ->assertResponseStatus(200)
+        ;
+    }
+
+    /** @test */
+    public function testUpdateRouteGate()
+    {
+        // Create a random registration with our centre.
+        $registration = factory(App\Registration::class)->create([
+            "centre_id" => $this->centre->id,
+        ]);
+
+        $edit_route = URL::route('service.registration.edit', [ 'registration' => $registration->id ]);
+        $login_route = URL::route('service.login');
+
+        // You can get there logged in.
+        $this->actingAs($this->user)
+            ->visit($edit_route)
+        ;
+
+        $this->type("changedByTest", "carer")
+            ->press("Save Changes")
+            ->seePageIs($edit_route)
+            ->assertResponseStatus(200)
+            ->seeElement("input[value=changedByTest]")
+        ;
+
+        Auth::logout();
+
+        $this->type("**blanked**", "carer")
+            ->press("Save Changes")
+            ->seePageIs($login_route)
+            ->assertResponseStatus(200)
+        ;
     }
```

Comment:
Do you also want to check that a user without permission to that centre cannot get past gate?

Reply:
Yes, will amend.

Reply:
Actually, won't amend in this PR - that feels like a test that can be added when i do the edit page tests.